docs/testnet-guardian-runbook.md

```diff
@@ -0,0 +1,130 @@
+4. Name of the release branch
+
+### Deploy
+
+Deploy a QA-net (currently a 200-node network) with the new binaries. Ensure gossip logs are enabled by setting the daemon flags `log-snark-work-gossip`, `log-txn-pool-gossip`, and `log-received-blocks`.
+
+Instructions for configuring and deploying a QA-net is roughly documented here: https://www.notion.so/codaprotocol/Testnet-deployment-notes-c1fbf8b8be9343b480f644c22d61ad85
+
+### Test
+
+After the QA-net is deployed, perform the following sanity-check:
+
+##### Sanity check
+
+1. Search for any crashes by querying the logs on StackDriver. The nodes are configured to restart upon crash for some of the known issues (#4295). Create issues for new ones.
+
+2. Run `scripts/ezcheck.sh` to check if all the nodes are synced and have the same state hash at their best tips.
+
+3. Query the logs to confirm that the nodes are creating and receiving blocks, transaction-pool diffs, and snark-pool diffs.
+Some strings to look for: `Received a block`, `Received Snark-pool diff`, `Received transaction-pool diff`
+
+4. Connect an external node to the network and check if it syncs
+(TODO: to automate these checks)
+
+Publish the peer ids of the seed nodes on `#QA-net details` channel for the rest of the team to connect to the QA-net.
+
+##### Complete the challenges for the release
+
+Connect to the QA-net from your machine and complete the protocol/product related challenges for the release. This is to test the features required for the challenges are working and also, may help the testnet guardian prepare for the kind of questions the community may have after the release.
+
+##### Check docs
+
+Verify that the docs are updated and user journeys are consistent with the release
+
+##### Monitor the QA-net and report status
+
+Monitor the health of the QA-net by checking the logs, grafana dashboard, telemetry and other tools. Check for
+1. Crashes
+2. Forks
+3. Expected TPS
+4. Blocks being produced consistently
+
+Report the status of the network and testing to the rest of the team on `#qa-net-discussion` channel
+
+On completing the QA-net phase successfully, let the infrastructure team know the same and tear down the QA-net before the release.
+
+Easy peasy?
+
+Maybe not, sometimes there could be unintended twists in the story and the testnet guardian might encounter issues along the way. Following is the list of possible situations and actions that the testnet guardian may take:
+
+| | Event | Action | Criticality | Owner(s)|
+|--|------|--------|-------------|--|
+|1| Unsuccessful build jobs. (We shouldn't be merging hotfixes before all the CI jobs are successful).| Sometimes rerunning the build job fixes it. For example, at times docker upload fails and passes on rerun. But if there are other code/ci config related issues then the QA-net cannot continue until the issue is either reverted or fixed. In the case of a hotfix for issues encountered during QA-net, the fix should be reverted| Blocker| |
```

Comment:
Hopefully this is mostly mitigated with #4768 , still worth calling it out as you did.